src/config.rs

```diff
@@ -269,6 +277,11 @@ impl Config {
             "top" => {
                 config.sampling_rate = value_t!(matches, "rate", u64)?;
             }
+            "completions" => {
+                let shell = value_t!(matches.value_of("shell"), clap::Shell).unwrap_or_else(|e| e.exit());
+                app.gen_completions_to(crate_name!(), shell, &mut std::io::stdout());
+                std::process::exit(0);
```

Comment:
I don't like this, but IMO `Config` is currently too tied with `clap::App`.

Reply:
In case you don't know, I'd recommend using [structopt](https://github.com/TeXitoi/structopt) instead of using `clap` directly.